File: eventx-frontend/src/services/ticketService.js
import api from './api';

const ticketService = {
  getTicket: (id) => api.get(`/tickets/${id}`),
  getTicketsForBooking: (bookingId) => api.get(`/tickets/booking/${bookingId}`),
  // Admin
  verifyTicket: (data) => api.post('/admin/tickets/verify', data),
};

export default ticketService;

File: eventx-backend/src/main/java/com/eventx/service/TicketService.java
package com.eventx.service;

import com.eventx.dto.request.VerifyTicketRequest;
import com.eventx.dto.response.TicketResponse;
import com.eventx.dto.response.TicketVerificationResponse;

import java.util.List;

public interface TicketService {
    List<TicketResponse> generateTickets(Long bookingId);
    TicketResponse getTicket(Long ticketId, Long userId);
    List<TicketResponse> getTicketsForBooking(Long bookingId, Long userId);
    TicketVerificationResponse verifyTicket(VerifyTicketRequest request);
}
